framework/auth/core.py

```diff
@@ -1345,8 +1348,63 @@ def n_projects_in_common(self, other_user):
         """Returns number of "shared projects" (projects that both users are contributors for)"""
         return len(self.get_projects_in_common(other_user, primary_keys=True))
 
+    def has_inst_auth(self, inst):
+        if inst in self.affiliated_institutions:
+            return True
+        return False
+
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
 
 def _merge_into_reversed(*iterables):
     '''Merge multiple sorted inputs into a single output in reverse order.
     '''
     return sorted(itertools.chain(*iterables), reverse=True)
+
+
+class Institution(StoredObject):
+
+    _id = fields.StringField(index=True, unique=True, primary=True)
```

Comment:
You want to set `default=lambda: str(ObjectId())`, or an `_id` won't get assigned on save. `ObjectId` can be imported from either `framework.mongo` or `bson`.